drivers/src/ecc384.rs

```diff
@@ -286,11 +290,16 @@ impl Ecc384 {
         let verify_r = Array4x12::read_from_reg(ecc.verify_r());
 
         // compare the hardware generate `r` with one in signature
-        if verify_r == signature.r {
-            Ok(true)
-        } else {
-            Ok(false)
-        }
+        let result = verify_r == signature.r;
+
+        self.zeroize();
+
+        Ok(result)
+    }
+
+    /// Zeroize the hardware registers.
+    pub fn zeroize(&mut self) {
+        self.ecc.regs_mut().ctrl().write(|w| w.zeroize(true));
```

Comment:
Does the zeroize operation really happen in a single cycle? Do we need to wait for this bit to clear before we can use the ECC peripheral again?

Reply:
As best as I can tell from the RTL, this doesn't seem to be a state machine.

Reply:
It's a one cycle operation; this has been confirmed with the hardware team.